thirdeye/thirdeye-frontend/app/pods/rca/details/controller.js

```diff
@@ -341,7 +341,14 @@ export default Controller.extend({
           shouldReset: false
         });
       }
-    }
+    },
 
+    /**
+     * Redirects the user to the new rootcause
+     * @param {Number} metricId
+     */
+    onLegacyToggle(metricId) {
+      this.send('transitionToNewRca', metricId);
```

Comment:
I think using bubbling through this.send is not very performant. Would it not be easier to just call that function directly?

Reply:
It's a route action. Do we have a better alternative? @cptran777